mudita/mudita-center
# [CP-272] Integrate tool with Center

Jira: [CP-272]

**Description**

This PR mainly implements the process described in the [Sent Pure Diagnostic Data (..).Workflow diagram](https://appnroll.atlassian.net/wiki/spaces/CP/pages/1487962167/Sent+Pure+Diagnostic+Data+-+high+order+abstraction#Workflow-diagram) chapter

**How to test?**
This story requires a new handler in [Mudita Center Server](https://github.com/mudita/mudita-center-server) so to test it, please wait until this [PR#8](https://github.com/mudita/mudita-center-server/pull/8) will be merge
or if you would like to, you can test MC with [Mudita Center Server](https://github.com/mudita/mudita-center-server) locally or use preview  from [PR#8](https://github.com/mudita/mudita-center-server/pull/8)
- before build the production application change `MUDITA_CENTER_SERVER_URL` value to the correct url

**Steps to test**
- Build production application
- Clear `settings.json` file:  `~/Library/Application\ Support/@mudita/mudita-center-app/settings.json`
- Run application
- Agree to submit diagnostic data
- Connect the pure to desktop 
- After the connection the logs from your pure should be
  - Downloaded
  - Sent to the pure logs storage server
    - Please write me to check that your logs vi serial nubmer are stored on the server 
  - In settings.json:
    - diagnosticSentTimestamp should be equal to the moment of connected device

**Final step: View the logs on the sftp server**
**Goal:** you are able to check the sent logs by following the steps according to the instruction file ([task](https://appnroll.atlassian.net/browse/CP-273): in progress)
**At the moment:** share me your device phone serial number, I will check that the files in storage


<details>
<summary><b>Screenshots</b></summary>

Logs from my device
<img width="504" alt="Zrzut ekranu 2021-07-19 o 16 30 54" src="https://user-images.githubusercontent.com/17147149/126178031-5aae4bac-47e9-452c-8be1-5989c51f8f6f.png">


</details>

**Self check**

- [x] Self CR'd
- [x] Description and screenshots attached

**PR Status**


**Blockers**

**Deploy Notes**


[CP-272]: https://appnroll.atlassian.net/browse/CP-272
